### ISSUE TITLE
Feature Request: Scheduled Imports / Folder Watch

### ISSUE THREAD

**Issue by sagansapien** [NUWCDIVNPT/stig-manager#53]
We use evaluate-STIG to constantly generate new CKLs, saved to a shared folder. This folder also contains manual CKLs not produced by evaluate-STIG.

These CKLs are being automatically created at any given moment, but manual action must be taken to import them to STIG Manager.

Would like to see feature that can import CKLs recursively from a specified folder path on a schedule or by watching for changes. This would allow STIG Manager to be fired up at any time and have the most recent CKLs from the scan result repository already parsed and imported.

Thanks!

**Comment by cd-rite** [comment]
This would be a really great candidate for a stand-alone utility that makes use of our API.  
We've re-architecting STIG Manager from it's "classic" incarnation to one with a documented RESTful API specifically to make it easier to do things like this.
Providing a feature like this on the STIGMan side would be fraught with user permission considerations, either setting up shares on the STIGMan server for people to submit checklists , or providing STIGMan with access to a set of shared folders, etc.  It could also get messy if more than one organization was using a central STIGMan instance, etc. 

However, a local client that watches a folder, and submits checklists (or, ideally, processes checklists into JSON for more efficient submission) when it sees them, or on a schedule, would be much easier to manage and customize for an Organizations needs. Collection Owners could also control access by adding or removing grants to this tools "user" to affect what assets they want it to update, etc. 

Long story short, I don't see a feature like this becoming part of STIG Manager itself, but something similar would definitely be useful and we would link so hard to the repository that had such a utility! Thanks!

**Comment by csmig** [comment]
To summarize, this is a case where a sidecar proxy is the answer. The functionality should not be part of the core API. I am finishing an example of such a utility and will push shortly.

**Comment by sagansapien** [comment]
Thank you for your consideration!
